### ISSUE TITLE
view-only wallet with imported key-image fails to show outgoing transfer

### ISSUE THREAD

**Issue by zveda** [monero-project/monero#1740]
I have a view-only wallet as well as a spending version of the same wallet. I then sweep all the monero from this wallet to another separate wallet. The spending version shows all the incoming and out-going transfers of the wallet. I then copy the key-images from the spending version to the view-only wallet. It now correctly shows the balance as 0 but it fails to show any outgoing transfers from the wallet. When I run show_transfers all I see is incoming transfers. Yet balance is 0. 

I am pretty sure this is not the intended behaviour. I would like to see any outgoing transfers from the view-only wallet with the inputs for which I have imported key-images. 

I am running Monero 'Wolfram Warptangent' (v0.10.1.0-release).

**Comment by Maxithi** [comment]
Does this still occur in the latest release?

**Comment by zveda** [comment]
Yes, I just checked. I exported key-images from the spending wallet and imported them in the watch-only. The balance shown is the same in both but the show_transfers are different. Some _out_ transactions are shown in the watch-only wallet and some are not. 

The final _out_ transaction in the spending wallet is shown as an _in_ transaction in the watch only, where the amount is the total balance of the wallet.

I'm using version 0.11.1.0.

**Comment by stoffu** [comment]
That's strange, I've made a patch that specifically addressed that exact issue: https://github.com/monero-project/monero/pull/2377

This should be included in version 0.11.1.0. Can you do `rescan_bc` on the watch-only wallet and try again? Also, can you post a full log?

**Comment by dEBRUYNE-1** [comment]
+bug

**Comment by zveda** [comment]
Just finished doing the rescan_bc, then re-imported the old key-images file (from yesterday). The balance is way too large. Seemed to add up all transactions as if they were _in_ transactions.

Then I exported key images file from the write wallet and imported it into the read-only wallet. Now read-only wallet shows the correct balance, but show-transfers still shows only _in_ transactions.

Attached log level 4 file from wallet, but I only turned the logging on for the last part of the rescan_bc and the imports of key-images. It is almost 20mb.
[monero-wallet-cli.log](https://github.com/monero-project/monero/files/1614004/monero-wallet-cli.log)

[bitmonero.log](https://github.com/monero-project/monero/files/1614014/bitmonero.log)
